### PR TITLE
fix compilation errors and minor deprecations for Android 13

### PR DESCRIPTION
**Pull request by dzinad**

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/service/NavigationNotificationService.kt [hunk 1]
```diff
@@ -43,7 +44,12 @@ internal class NavigationNotificationService : Service() {
      */
     override fun onDestroy() {
         super.onDestroy()
-        stopForeground(true)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            stopForeground(STOP_FOREGROUND_REMOVE)
+        } else {
+            @Suppress("DEPRECATION")
+            stopForeground(true)
+        }
```
Comment (tomaszrybakiewicz) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/service/NavigationNotificationService.kt:52]:
```suggestion
        ServiceCompat.stopForeground(this, ServiceCompat.STOP_FOREGROUND_REMOVE)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,9 @@ Mapbox welcomes participation and contributions from everyone.
 - Added `guideMapUri` to the `RestStop`. [#6237](https://github.com/mapbox/mapbox-navigation-android/pull/6237)
 - [TileStore Android Service] Fixed a crash when the service process is killed by the Android system. [#6237](https://github.com/mapbox/mapbox-navigation-android/pull/6237)
 - Fixed `MapboxNavigationApp#detach` will not fully detach. This causes `MapboxNavigation` to continue to be accessible, and causes `MapboxNavigationObserver.onDetached` to be called multiple times.
+- Added Android 13 support. [#6196](https://github.com/mapbox/mapbox-navigation-android/pull/6196)
+- Declared [POST_NOTIFICATIONS](https://developer.android.com/reference/android/Manifest.permission#POST_NOTIFICATIONS) permission in SDK's AndroidManifest.xml. It is highly recommended for apps to request the permission in runtime, Without it the SDK will not be able to show the notification with trip progress in the notification drawer for apps that target Android 13 or higher. [#6196](https://github.com/mapbox/mapbox-navigation-android/pull/6196)
```
Comment (LukasPaczos) [inline CHANGELOG.md:14]:
```suggestion
- Declared [POST_NOTIFICATIONS](https://developer.android.com/reference/android/Manifest.permission#POST_NOTIFICATIONS) permission in SDK's AndroidManifest.xml. It is highly recommended for apps to request the permission in runtime. Without it, the SDK will not be able to show the notification with trip progress in the notification drawer for apps that target Android 13 or higher. [#6196](https://github.com/mapbox/mapbox-navigation-android/pull/6196)
```
nit, can be addressed during release

**Review by LukasPaczos — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,9 @@ Mapbox welcomes participation and contributions from everyone.
...
+- Bumped targetSdkVersion to 33. [#6196](https://github.com/mapbox/mapbox-navigation-android/pull/6196)
```
Comment (LukasPaczos) [inline CHANGELOG.md:15]: Is this actually relevant to app developers? The app manifest has a higher priority and its `targetSdkVersion` will be chosen in the manifest merger either way.

Comment (dzinad) [reply CHANGELOG.md]: Removed it.
